fraqbot/Local/randomemoji.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+import random
+import sys
+
+from Legobot.Lego import Lego
+
+
+logger = logging.getLogger(__name__)
+LOCAL_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)))
+
+if LOCAL_DIR not in sys.path:
+    sys.path.append(LOCAL_DIR)
+
+
+from helpers import utils  # noqa: E402
+
+
+class RandomEmoji(Lego):
+    def __init__(self, baseplate, lock, *args, **kwargs):
+        super().__init__(baseplate, lock, acl=kwargs.get('acl'))
+
+    def listening_for(self, message):
+        text = message.get('text')
+
+        return isinstance(text, str) and text.startswith('!emoji')
+
+    def _get_emoji(self, how_many):
+        maximum_how_many = 20
+        minimum_how_many = 1
+        how_many_limited = max(minimum_how_many, min(maximum_how_many, how_many))
+        
+        emoji_response = utils.call_slack_api(
+                self.slack_client,
+                'emoji.list',
+                True,
+                'emoji',
+                total_limit=10000,
+                limit=1000,
+            )
```

Comment:
the call_slack_api assumes the response after transform is a list, but this api call returns a dict. Good news is it's also not paginated, so I've tested and it works. You don't need to include the limits and you can set get_all to False rather than True.